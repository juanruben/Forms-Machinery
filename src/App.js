import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

import Login from './Pages/Login';
import Home from './Pages/Home';

class App extends Component {
	render() {
		return (
        <Router>
            <Switch> 
              <Route exact path="/" component={Login}  />
              <Route exact path="/Home" component={Home}  />
            </Switch>
        </Router>
    )
	}
}

export default App;