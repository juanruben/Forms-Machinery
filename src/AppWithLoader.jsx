import React from 'react';
import { useStateValue } from './State';
import Loader from './Components/Loader/Loader';
import AppRouter from './AppRouter';

function AppWithLoader() {
    const [{ loading }] = useStateValue();
    return (
        <Loader visible={loading}>
            <AppRouter />
        </Loader>
    );
}

export default AppWithLoader;
