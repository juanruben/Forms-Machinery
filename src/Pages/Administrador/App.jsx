import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Home from './Components/Home';
import RegistroHistorico from './Components/RegistroHistorico';
import ClientLists from './Components/Client/ClientLists';
import ClientAdd from './Components/Client/ClientAdd';
import UserLists from './Components/User/UserLists';
import UserAdd from './Components/User/UserAdd';
import FormAdd from './Components/Form/FormAdd';
import FormLists from './Components/Form/FormLists';
import MachineryLists from './Components/Machinery/MachineryLists';
import MachineryAdd from './Components/Machinery/MachineryAdd';

const App = () => (
  <Router>
    <Switch>
      <Route exact path="/administrador/home/" component={Home} />
      <Route exact path="/administrador/registro-historico/" component={RegistroHistorico} />
      <Route exact path="/administrador/cliente/add/" component={ClientAdd} />
      <Route exact path="/administrador/cliente/lists/" component={ClientLists} />
      <Route exact path="/administrador/usuario/add/" component={UserAdd} />
      <Route exact path="/administrador/usuario/lists/" component={UserLists} />
      <Route exact path="/administrador/maquina/add/" component={MachineryAdd} />
      <Route exact path="/administrador/maquina/lists/" component={MachineryLists} />
      <Route exact path="/administrador/formulario/add/" component={FormAdd} />
      <Route exact path="/administrador/formulario/lists/" component={FormLists} />
    </Switch>
  </Router>
);

export default App;
