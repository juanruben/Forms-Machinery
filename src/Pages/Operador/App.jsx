import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Home from './Components/Home';

const App = () => (
  <Router>
    <Switch>
      <Route exact path="/operador/home/" component={Home} />
    </Switch>
  </Router>
);
export default App;
