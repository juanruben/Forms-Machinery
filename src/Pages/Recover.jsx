import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import {
  Container,
  Row,
  Col,
  Form,
  FormGroup,
  Button,
  Alert,
} from 'reactstrap';
import Api from '../Service/Api';
import Logo from '../Assets/images/Logo.png';

class Recover extends Component {
  constructor(props) {
    super(props);
    this.state = {
      inputEmail: '',
      textMsj: '',
      alertClass: '',
      disabledButton: false,
    };
    this.handleInputChange = this.handleInputChange.bind(this);
    this.procesar = this.procesar.bind(this);
  }

  async processInformation() {
    const dataProcesar = {};
    const { inputEmail } = this.state;
    dataProcesar.email = inputEmail;
    const responseApi = await Api.RecuperarContrasenia(dataProcesar);
    if (responseApi.result === 'success') {
      console.info('Contraseña fue recuperada coractamente');
    } else {
      this.setState({
        textMsj: 'Ocurrió un prolema al intentar recuperar contraseña',
        alertClass: 'alert-danger',
      });
    }
  }

  handleInputChange(event) {
    this.setState({ textMsj: '' });
    const { value, name } = event.target;
    this.setState({ [name]: value });
    const { email } = this.state;
    if (email !== '') {
      this.setState({ disabledButton: true });
    } else {
      this.setState({ disabledButton: false });
    }
  }

  render() {
    const {
      textMsj,
      alertClass,
      inputEmail,
      disabledButton,
    } = this.state;

    return (
      <Container>
        <Row>
          <Col md={{ size: 6, offset: 3 }}>
            <div className="container-login box">
              <div className="container__logo">
                <img src={Logo} alt="" />
              </div>
              <Form>
                <FormGroup row>
                  <Col sm={12}>
                    <input type="text" name="email" id="email" onChange={this.handleInputChange} value={inputEmail || ''} placeholder="Email" className="form-control" />
                  </Col>
                </FormGroup>
                <FormGroup row>
                  <Col sm={12} className="text-center">
                    {textMsj !== '' ? <Alert className={`alert ${alertClass}`}>{textMsj}</Alert> : '' }
                  </Col>
                </FormGroup>
                <FormGroup row>
                  <Col sm={12} className="text-center">
                    <Button onClick={this.processInformation} className="btn-orange" disabled={disabledButton}>Enviar</Button>
                  </Col>
                </FormGroup>
                <FormGroup row>
                  <Col sm={12} className="text-center">
                    <Link to="/" className="link">Volver</Link>
                  </Col>
                </FormGroup>
              </Form>
            </div>
          </Col>
        </Row>
      </Container>
    );
  }
}

export default Recover;
