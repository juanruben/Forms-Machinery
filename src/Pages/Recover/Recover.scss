.message-ok{
	color: rgb(0, 85, 0);
	font-weight: bold;
	padding-top: 10px;
}