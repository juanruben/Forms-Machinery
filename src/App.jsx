import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Login from './Pages/Login';
import Recuperar from './Pages/Recover';
import Administrador from './Pages/Administrador/App';
import Operador from './Pages/Operador/App';

const App = () => (
    <Router>
        <Switch>
            <Route exact path="/" component={Login} />
            <Route exact path="/Recuperar" component={Recuperar} />
            <Route path="/administrador/" component={Administrador} />
            <Route path="/operador/" component={Operador} />
        </Switch>
    </Router>
);

export default App;
