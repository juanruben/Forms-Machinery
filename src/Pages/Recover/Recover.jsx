import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { StateContext } from '../../State';
import { validateEmail } from '../../Service/Utils';
import LayoutFullWidth from '../../Layout/LayoutFullWidth/LayoutFullWidth';
import Box from '../../Layout/Box/Box';
import Logo from '../../Components/Logo/Logo';
import Input from '../../Components/Input/Input';
import Button from '../../Components/Button/Button';

import './Recover.scss';

class Recover extends Component {
    constructor(props) {
        super(props);
        this.state = {
            email: '',
            errors: {},
            sent: false,
        };
        this.handleInputChange = this.handleInputChange.bind(this);
        this.validForm = this.validForm.bind(this);
        this.send = this.send.bind(this);
    }

    validForm() {
        const { email } = this.state;
        const errors = {};
        let formIsValid = true;

        if (!validateEmail(email)) {
            formIsValid = false;
            errors.email = 'Error de formato de email';
        }

        this.setState({
            errors,
        });

        return formIsValid;
    }

    toggleLoading(value) {
        const [, dispatch] = this.context;
        dispatch({
            type: 'SET_LOADING',
            value,
        });
    }

    async send() {
        if (this.validForm()) {
            this.toggleLoading(true);
            // const { email } = this.state;

            // const response = await LoginApi({
            //     username,
            //     password,
            // });
            this.setState({
                sent: true,
            });

            // if (responseApi.result === 'success') {
            //     if (responseApi.perfil === 1) {
            //         window.location.assign('/administrador/');
            //     } else {
            //         window.location.assign('/operador/');
            //     }
            // }

            this.toggleLoading(false);
        }
    }

    handleInputChange(event) {
        const { name, value } = event.target;
        this.setState({
            [name]: value,
            errors: {},
        });
    }

    render() {
        const {
            email,
            errors,
            sent,
        } = this.state;

        return (
            <LayoutFullWidth>
                <Box>
                    <Logo padding={30} maxWidth={150} />
                    <Input type="email" name="email" onChange={this.handleInputChange} value={email} icon="far fa-envelope" placeholder="Correo electrónico" errors={errors.email} />
                    <Button type="button" onClick={this.send} text="Enviar" />
                    <Link to="/" className="link-login">Volver</Link>
                </Box>
                {sent && <div className="message-ok">Revisa tu correo y verifica tu contraseña</div>}
            </LayoutFullWidth>
        );
    }
}

Recover.contextType = StateContext;

export default Recover;
