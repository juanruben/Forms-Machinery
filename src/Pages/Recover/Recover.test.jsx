import React from 'react';
import { mount, shallow } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { StateProvider } from '../../State';
import { initialState, reducer } from '../../App';
import AppWithLoader from '../../AppWithLoader';
import Recover from './Recover';
import Button from '../../Components/Button/Button';

jest.mock('./../../Service/Api');

describe('Recover', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Recover />);
    });

    it('should render correctly', () => expect(wrapper.debug()).toMatchSnapshot());

    it('should test ', () => {
        const mockReducer = jest.fn();
        const wrapper2 = mount(
            <StateProvider initialState={initialState} reducer={mockReducer}>
                <Recover />
            </StateProvider>,
        );
        const wrapperRecover = wrapper2.find(Recover).at(0);
        act(() => {
            wrapperRecover.instance().toggleLoading(false);
        });
        wrapper2.update();
        expect(mockReducer).toHaveBeenCalledTimes(1);
        wrapper2.unmount();
    });

    it('should call validate when send', () => {
        const mockValidate = jest.fn();
        wrapper.instance().validForm = mockValidate;
        wrapper.instance().handleInputChange({ target: { name: 'email', value: 'test@test.com' } });
        wrapper.instance().send();
        expect(mockValidate).toHaveBeenCalledTimes(1);
    });


    it('should show and hide loading', () => {
        const mockToggleLoading = jest.fn();
        wrapper.instance().toggleLoading = mockToggleLoading;
        wrapper.instance().handleInputChange({ target: { name: 'email', value: 'test@test.com' } });
        wrapper.find(Button).props().onClick();
        expect(mockToggleLoading).toHaveBeenCalledTimes(2);
    });

    it('should update state', () => {
        wrapper.instance().handleInputChange({ target: { name: 'email', value: 'test@test.com' } });
        expect(wrapper.state('email')).toEqual('test@test.com');
    });

    it('should validate', () => {
        wrapper.setState({ email: 'test@test.com' });
        expect(wrapper.instance().validForm()).toBe(true);
        wrapper.setState({ email: '' });
        expect(wrapper.instance().validForm()).toBe(false);
    });

    it('should have sent message', () => {
        wrapper.setState({ sent: true });
        expect(wrapper.find('.message-ok')).toHaveLength(1);
    });

    it('should have a logo, two inputs and a button', () => {
        expect(wrapper.find('Logo')).toHaveLength(1);
        expect(wrapper.find('Input')).toHaveLength(1);
        expect(wrapper.find('Button')).toHaveLength(1);
    });

    afterEach(() => wrapper.unmount());
});

// describe('Login process', () => {
//     it('should log in', (done) => {
//         const wrapper = mount(
//             <StateProvider initialState={initialState} reducer={reducer}>
//                 <AppWithLoader />
//             </StateProvider>,
//         );
//         const wrapperRecover = wrapper.find(Login).at(0);
//         const spyValidation = jest.spyOn(wrapperRecover.instance(), 'validForm');
//         setTimeout(() => {
//             act(() => {
//                 wrapper.update();
//                 wrapperRecover.setState({ username: 'test', password: 'Test1234' });
//                 wrapperRecover.instance().handleSignIn();
//             });
//             expect(spyValidation).toBeCalledTimes(1);
//             expect(wrapperRecover.instance().validForm()).toBeTruthy();
//             wrapper.update();
//             expect(wrapper.render().find('Main')).toBeTruthy();

//             wrapper.unmount();

//             done();
//         });
//     });
// });
