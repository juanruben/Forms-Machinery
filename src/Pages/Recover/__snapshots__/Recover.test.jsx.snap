// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recover should render correctly 1`] = `
"<LayoutFullWidth>
  <Box>
    <Logo />
    <Input type=\\"email\\" name=\\"email\\" onChange={[Function: bound handleInputChange]} value=\\"\\" icon=\\"far fa-envelope\\" placeholder=\\"Correo electrÃ³nico\\" errors=\\"\\" />
    <Button type=\\"submit\\" onClick={[Function: bound send]} text=\\"Enviar\\" disabled={false} />
  </Box>
</LayoutFullWidth>"
`;
